### PR TITLE
[JENKINS-42724] - Restore the windows-service/jenkins.xml resource to restore compatibility with windows-slaves 1.2

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Background:

> Windows Slaves plugin 1.0 performs a direct access to the resources bundled into the core.
Hence the file removal was a bad idea though I have not seen the issue in automatic tests and ATH. This change also was a last-minute change in https://github.com/jenkinsci/jenkins/pull/2765/ in order to address suggestions from @daniel-beck, hence I didn't test it properly. Sorry about that.

See [JENKINS-42724](https://issues.jenkins-ci.org/browse/JENKINS-42724) and [JENKINS-42746](https://issues.jenkins-ci.org/browse/JENKINS-42746).

No autotests, it should be somehow caught by PCT or ATH.

[Full diff](https://github.com/jenkinsci/windows-slaves-plugin/compare/windows-slaves-1.0...windows-slaves-1.3.1)

### Changelog entries

Proposed changelog entries:

* bug - JENKINS-42724: Restore the deprecated `windows-service/jenkins.xml` resource to restore the compatibility with Windows Agents Plugin (FKA Windows Slaves Plugin) 1.2 and below
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Link to JIRA ticket in description, if appropriate
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc)

### Desired reviewers

@daniel-beck @jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Review by abayer — DISMISSED** [review]

File: core/src/main/resources/windows-service/jenkins-slave.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  -->
+  <logmode>rotate</logmode>
+
+  <onfailure action="restart" />
+  
+  <!-- 
+    In the case WinSW gets terminated and leaks the process, we want to abort
+    these runaway JAR processes on startup to prevent "Slave is already connected errors" (JENKINS-28492).
+  -->
+  <extensions>
```
Comment (jglick) [inline core/src/main/resources/windows-service/jenkins-slave.xml]: So this is copied from `jenkins.xml`, and was not in the deleted `jenkins-slave.xml`, right?

File: core/src/main/resources/windows-service/jenkins-slave.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    <!-- This is a sample configuration for the RunawayProcessKiller extension. -->
+    <extension enabled="true" 
+               className="winsw.Plugins.RunawayProcessKiller.RunawayProcessKillerExtension"
+               id="killOnStartup">
+      <pidfile>%BASE%\jenkins_agent.pid</pidfile>
+      <stopTimeout>5000</stopTimeout>
```
Comment (jglick) [inline core/src/main/resources/windows-service/jenkins-slave.xml]: Except 5000 rather than 10000?

**Review by jglick — DISMISSED** [review]

File: core/src/main/resources/windows-service/jenkins-slave.xml (hunk 1 above)
Comment (oleg-nenashev) [inline core/src/main/resources/windows-service/jenkins-slave.xml]: This is copied from https://github.com/jenkinsci/windows-slave-installer-module/blob/master/src/main/resources/org/jenkinsci/modules/windows_slave_installer/jenkins-slave.xml .

I cannot enable the automatic update easily due to the Copy-paste of the old stuff in Windows Slaves plugin, but at least I can pick the new sample for other options

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
Also created a workaround PR to the plugin: https://github.com/jenkinsci/windows-slaves-plugin/pull/6 . With that change we may avoid the out-of-order release

**Comment by oleg-nenashev** [comment]
Should I merge it to formally close the regression? Likely yes, because there will be more people updating to 2.50+ without updating to Windows Slaves 1.3.1+

**Comment by daniel-beck** [comment]
Isn't that just a matter of upgrading the bundled version?

**Comment by oleg-nenashev** [comment]
> Isn't that just a matter of upgrading the bundled version?

Well, maybe. I doubt there are any other usages of this file. And [Windows Slaves Changelog](https://wiki.jenkins-ci.org/display/JENKINS/Windows+Slaves+Plugin) does not contain potential fatal flaws. Only minor changes since the decoupling

**Comment by daniel-beck** [comment]
In that case I would favor that variant.

**Comment by oleg-nenashev** [comment]
@reviewbybees @daniel-beck I need another review iteration since I have changed the implementation after the release of 1.3.1 and feedback from @daniel-beck

**Review by daniel-beck — APPROVED** [review]
Assuming this fixes the issue for people who had an older version installed, i.e. it automatically is upgraded.

Also, should be squashed.

**Comment by oleg-nenashev** [comment]
> Assuming this fixes the issue for people who had an older version installed, i.e. it automatically is upgraded.

Will check it, but I have some concerns about that

**Comment by oleg-nenashev** [comment]
@daniel-beck Checked, it does not work OOTB. So I will restore the resource file

**Comment by oleg-nenashev** [comment]
@daniel-beck fixed

**Comment by daniel-beck** [comment]
As discussed, :+1: conditional on followup investigation, as #2489 etc. indicate raising the bundled version should upgrade every older, installed plugin.

**Comment by oleg-nenashev** [comment]
Note: I've just got an issue during the upgrade from 1.642. The restart should have fixed the issue, but still a potential case.

As discussed with @daniel-beck , we agreed not to bump the dependency now since the detached plugin needs to be bumped accordingly in https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/ClassicPluginStrategy.java#L413...L427 . And @daniel-beck thinks it is not trivial in this case

**Comment by oleg-nenashev** [comment]
Merging to get it in the release

**Pull request merged**